base/src/main/java/uk/ac/ebi/atlas/solr/query/SolrQueryService.java

```diff
@@ -74,4 +72,15 @@ public GeneQueryResponse fetchResponse(SemanticQuery geneQuery, Species species)
 
         return geneQueryResponse;
     }
+
+    public Set<String> fetchSpecies(SemanticQueryTerm geneQueryTerm) {
```

Comment:
This is confusing and doesn't look correct: why is the default property name "bioentity_identifier" ? If the category needs to be there, then why does the method take a SemanticQueryTerm that doesn't have to have a category?
Why are we not always delegating to `SemanticQueryTerm::asGxaIndexQueryLiteral` and adding this special case here?

Reply:
The problem is that if you look for, say, `ENSMUSG00000019082`, and use `asGxaIndexQueryLiteral` you will search in `property_value` and the first document that matches is:
```
{
        "bioentity_identifier":"WBGene00018901",
        "species":"Caenorhabditis_elegans",
        "property_value":"ENSMUSG00000019082",
        "property_weight":0,
        "property_name":"ortholog",
        "id":"8448a632-94b3-49e1-8bf3-dc95e6d8edd8",
        "_version_":1571896862507532314},
```

Then the species is resolved as *Caenorhabditis elegans*. This is specified in the the test `SolrQueryServiceTest::getKnownSpeciesWithoutCategory`.

The category isn’t mandatory, it may or may not be there, but if it isn’t what you want is to query `bioentity_identifier`. The one change I’m adding is to search first in `bioentity_identifier` and falling back to the regular `asGxaIndexQueryLiteral` if no results are returned.